### DISCUSSION TITLE
Compatibility Only Until Python 3.10 - Python 3.10 is no longer installable

### DISCUSSION THREAD

**Discussion by aboushawareb-mariem** [scikit-activeml/scikit-activeml#411] [Q&A]
Hello everyone,

I'm trying to install this library, however, I kept getting this error while pip installing: 
![image](https://github.com/scikit-activeml/scikit-activeml/assets/16010276/f3e2ea68-180d-4324-a5cc-f730bfa4a095)

Then I realized that the Python version specified for this library is only until Python 3.10
![image](https://github.com/scikit-activeml/scikit-activeml/assets/16010276/172f349a-4cb7-44d6-b9b8-9bdf98e57f1a)

I tried to downgrade to Python 3.10 from my installed 3.12, but I think Python 3.10 is no longer available for installation: https://www.python.org/downloads/release/python-31014/

![image](https://github.com/scikit-activeml/scikit-activeml/assets/16010276/59440d9c-7307-484f-8761-76bbb90d5a2f)

Is there a work around this? I think this way the library is not usable anymore. Thanks a lot for your help!!

**Comment by mherde** [comment]
Hi @aboushawareb-mariem 

Thanks for your interest in using our active learning library `scikit-activeml`. Indeed, not all Python versions have a downloadable installer. However, there are multiple Python versions that conform to our library's current requirements, e.g., 

- [Python 3.10](https://www.python.org/downloads/release/python-31011/),
- [Python 3.9](https://www.python.org/downloads/release/python-3912/),

and a full overview of available installers for Windows as an exemplary operating system is given [here](https://www.python.org/downloads/windows/).

Furthermore, you can also install [`conda`](https://conda.io/projects/conda/en/latest/user-guide/install/index.html), which allows you to create so-called [`conda` environments](https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html) with a specific Python version, e.g.,
```python
conda create -n myenv python=3.10
conda activate myenv
conda install pip
pip install -U scikit-activeml
```

We plan to publish a new release supporting recent Python versions within the upcoming weeks. However, the above options still allow the usage of our library. 

We hope that answers your question and allows you to install our library. If so, feel free to close this discussion. Otherwise, please ask further questions, and we will reply as fast as possible.

Best regards
Marek

**Comment by aboushawareb-mariem** [comment]
Thank you @mherde I actually found the installer for 3.10 yesterday but I still face issues while installing:

![image](https://github.com/scikit-activeml/scikit-activeml/assets/16010276/2dff3042-5094-48f2-9502-66eab563d9d5)


Would you be able to help me with this? I tried updating the wheel package or purging my cache, but these solutions don't work. I noticed that on Google Colab it works but on my machine it doesn't. Does it have to do with Windows vs Linux machine? Any suggestions on how to make it work? Thanks a lot!

**Comment by tpham93** [comment]
Hello @aboushawareb-mariem,

the problem you are facing seems to be a general problem with your environment. As the error message says, you are missing the tools to compile C++ code which is required by our dependencies. For this, you need to download and install Visual CPP Build tools, as shown in the error message.

Furthermore, I agree with @mherde that using a conda installation to manage Python environments in Windows is advisable as I use it personally as well.

Best regards
Tuan

**Comment by mherde** [comment]
Hi @aboushawareb-mariem,

Did our suggestions work so that you could install `scikit-activeml`?

Best regards
Marek

**Reply by aboushawareb-mariem** [reply]
Hello ..Not really since I couldn't use Anaconda for other reasons but I managed to fix the error by setting this environment variable PlatformToolset to my installed Visual Studio Build Tools version .. example for version 17.10.4 is PlatformToolset=v17104. Then it works :) In case anyone else was in the same situation. 

Thank you @mherde and @tpham93 for your prompt responses 🚀 ^^
